common/video-tiles.c

```diff
@@ -1,143 +1,115 @@
 #include "video-tiles.h"
 #include "minefield.h"
 
-void draw_single_cell(minefield* mf, uint8_t x, uint8_t y){
-	if (CELL(mf, x, y) & ISOPEN) {
-		if (CELL(mf, x, y) & HASBOMB){
-			set_tile(MINEFIELD_X_OFFSET + x*2 + 1,
-					 MINEFIELD_Y_OFFSET + y*2 + 1,
-					 BOMB);
-		} else {
-			uint8_t tile_number = 0;
-			uint8_t count = CELL(mf, x, y) & 0x0F;
-			if (count > 0 && count < 9) {
-				tile_number = ONE_BOMB + count - 1;
-				set_tile(MINEFIELD_X_OFFSET + x * 2 + 1, 
-				         MINEFIELD_Y_OFFSET + y * 2 + 1,
-				         tile_number);
-			} else {
-				set_tile(MINEFIELD_X_OFFSET + x * 2 + 1,
-				         MINEFIELD_Y_OFFSET + y * 2 + 1,
-				         BLANK);
-			}
-		}
-	} else {
-		if (CELL(mf, x, y) & HASFLAG){
-			set_tile(MINEFIELD_X_OFFSET + x*2 + 1,
-					 MINEFIELD_Y_OFFSET + y*2 + 1,
-					 FLAG);
-		} else if (CELL(mf, x, y) & HASQUESTIONMARK){
-			set_tile(MINEFIELD_X_OFFSET + x*2 + 1,
-					 MINEFIELD_Y_OFFSET + y*2 + 1,
-					 QUESTION_MARK);
-		} else {
-			set_tile(MINEFIELD_X_OFFSET + x*2 + 1,
-					 MINEFIELD_Y_OFFSET + y*2 + 1,
-					 CLOSED_CELL);
-		}
-	}
+void draw_single_cell(minefield *mf, uint8_t x, uint8_t y)
+{
+    const uint8_t cell = CELL(mf, x, y);
+
+    x = (x << 1) + MINEFIELD_X_OFFSET + 1;
+    y = (y << 1) + MINEFIELD_Y_OFFSET + 1;
+
+    if (cell & ISOPEN) {
+        if (cell & HASBOMB) {
+            set_tile(x, y, BOMB);
+        } else {
+            uint8_t count = cell & 0x0F;
+            if (count > 0 && count < 9) {
+                set_tile(x, y, ONE_BOMB + count - 1);
+            } else {
+                set_tile(x, y, BLANK);
+            }
+        }
+    } else {
+        if (cell & HASFLAG) {
+            set_tile(x, y, FLAG);
+        } else if (cell & HASQUESTIONMARK) {
+            set_tile(x, y, QUESTION_MARK);
+        } else {
+            set_tile(x, y, CLOSED_CELL);
+        }
+    }
```

Comment:
**Suggestion:** What if we only calculate the `tile_id` in this code block and then call `set_tile(x, y, tile_id)` only once at the end? Would that also reduce code-size?

Reply:
That might work well, yeah. Let me try this one.

Reply:
Eh, saved like 4 bytes :P